tools/crm_resource.c

```diff
@@ -1106,62 +1120,82 @@ main(int argc, char **argv)
                 g_hash_table_replace(options.override_params, name, value);
 
             } else {
-                CMD_ERR("Error parsing '%s' as a name=value pair for --%s", argv[optind], options.rsc_long_cmd);
+                g_set_error(&error, PCMK__EXITC_ERROR, CRM_EX_USAGE,
+                            "Error parsing '%s' as a name=value pair for --%s", argv[optind], options.rsc_long_cmd);
                 free(value);
                 free(name);
-                argerr++;
+                goto done;
             }
             optind++;
         }
 
     } else if (optind < argc && argv[optind] != NULL && options.rsc_cmd == 0) {
-        CMD_ERR("non-option ARGV-elements: ");
+        gchar **strv = calloc(argc-optind, sizeof(char *));
+        gchar *msg = NULL;
+        int i = 1;
+
+        strv[0] = strdup("non-option ARGV-elements:");
+
         while (optind < argc && argv[optind] != NULL) {
-            CMD_ERR("[%d of %d] %s ", optind, argc, argv[optind]);
+			strv[i] = crm_strdup_printf("[%d of %d] %s\n", optind, argc, argv[optind]);
             optind++;
-            argerr++;
+			i++;
```

Comment:
looks like some extra indenting here

Reply:
These patches have been rebased so many times.  Looks like a little of this snuck through, probably from a copy and paste.